### PR TITLE
[WIP][12.0][IMP] add by_method exception

### PR DESCRIPTION
**Pull request by sebastienbeau**
need to add the test, and finish it  @kevinkhao

### CODE REVIEW

**Comment by kevinkhao** [comment]
https://github.com/akretion/server-tools/pull/25

**Comment by sebastienbeau** [comment]
@bealdav @hparfr @florian-dacosta I think you can be interested by this improvement

**Review by hparfr — CHANGES_REQUESTED** [review]
It's a nice feature, but about the implementation, I suggest to limit methods to hard-coded known methods.

changing to `method = fields.Selection()` instead of 
`fields.Char` will do the trick.

`_check_method_valid` will be useless and it will be safer.

**Comment by sebastienbeau** [comment]
@hparfr great idea, thanks, Done ok for you ?

File: base_exception/models/base_exception.py [hunk 1]
```diff
@@ -53,10 +53,6 @@ def check_exception_type_consistency(self):
             if (
                 (rule.exception_type == "by_py_code" and not rule.code)
                 or (rule.exception_type == "by_domain" and not rule.domain)
-                or (
-                    rule.exception_type == "by_method"
-                    and not self._check_method_valid(rule.method, rule.model)
-                )
```
Comment (hparfr) [inline base_exception/models/base_exception.py]: Please add back a check on rule.method which can be undefined.
```py
 or (rule.exception_type == "by_method" and not rule.method)
```

Comment (sebastienbeau) [reply base_exception/models/base_exception.py]: Thanks for the feedback ! DONE

**Review by hparfr — APPROVED** [review]
Thanks

**Review by florian-dacosta — APPROVED** [review]

**Review by bealdav — APPROVED** [review]
LGTM

**Pull request merged**